src/tracer/propagation/cloud_trace_propagator.cpp

```diff
@@ -0,0 +1,213 @@
+  } else {
+    result = this->ExtractSpanContextImpl(carrier, trace_context, iequals, baggage);
+  }
+  if (!result || !*result) {
+    return result;
+  }
+
+  return result;
+}
+
+//--------------------------------------------------------------------------------------------------
+// InjectSpanContextImpl
+//--------------------------------------------------------------------------------------------------
+opentracing::expected<void> CloudTracePropagator::InjectSpanContextImpl(
+    const opentracing::TextMapWriter& carrier,
+    const TraceContext& trace_context) const {
+  std::array<char, CloudContextLength> buffer;
+  auto data_length = this->SerializeCloudTrace(trace_context, buffer.data());
+  return carrier.Set(PropagationSingleKey,
+                  opentracing::string_view{buffer.data(), data_length});
+}
+
+//--------------------------------------------------------------------------------------------------
+// ExtractSpanContextImpl
+//--------------------------------------------------------------------------------------------------
+template <class KeyCompare>
+opentracing::expected<bool> CloudTracePropagator::ExtractSpanContextImpl(
+    const opentracing::TextMapReader& carrier, TraceContext& trace_context, const KeyCompare& key_compare,
+    BaggageProtobufMap& baggage) const {
+      bool parent_header_found = false;
+      auto result =
+          carrier.ForeachKey([&](opentracing::string_view key,
+                                 opentracing::string_view
+                                     value) noexcept->opentracing::expected<void> {
+            if (key_compare(key, PropagationSingleKey)) {
+              auto was_successful = this->ParseCloudTrace(value, trace_context);
+              if (!was_successful) {
+                return opentracing::make_unexpected(was_successful.error());
+              }
+              parent_header_found = true;
+            } else if (key.length() > PrefixBaggage.size() &&
+                       key_compare(opentracing::string_view{key.data(),
+                                                            PrefixBaggage.size()},
+                                   PrefixBaggage)) {
+              baggage.insert(BaggageProtobufMap::value_type(
+                  ToLower(
+                      opentracing::string_view{key.data() + PrefixBaggage.size(),
+                                               key.size() - PrefixBaggage.size()}),
+                  value));
+            }
+            return {};
+          });
+      if (!result) {
+        return opentracing::make_unexpected(result.error());
+      }
+      return parent_header_found;
+}
+
+opentracing::expected<void> CloudTracePropagator::ParseCloudTrace(
+    opentracing::string_view s, TraceContext& trace_context) const noexcept {
+  if (s.size() < Num128BitHexDigits) {
+    return opentracing::make_unexpected(
+        std::make_error_code(std::errc::invalid_argument));
```

Comment:
This line's missing from test coverage:
https://8176-57146219-gh.circle-artifacts.com/0/coverage/tracer/propagation/cloud_trace_propagator.cpp.gcov.html

Reply:
added test case